### PR TITLE
fix(keys): fix range checks on input

### LINKED ISSUE #91: deis keys:add should not throw a hard error
```
╰─± deis keys:add
Found the following SSH public keys:
1) deis.pub deis
2) id_boot2docker.pub jonathanchauncey@ENG000637.local
3) id_rsa.pub jonathanchauncey@ENG000637.local
0) Enter path to pubfile (or use keys:add <key_path>)
Which would you like to use with Deis? 4
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/deis/workflow-cli/cmd.chooseKey(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jonathanchauncey/go/src/github.com/deis/workflow-cli/cmd/keys.go:130 +0xc43
github.com/deis/workflow-cli/cmd.KeyAdd(0x0, 0x0, 0x0, 0x0)
	/Users/jonathanchauncey/go/src/github.com/deis/workflow-cli/cmd/keys.go:71 +0xc6
github.com/deis/workflow-cli/parser.keyAdd(0xc820088010, 0x1, 0x1, 0x0, 0x0)
	/Users/jonathanchauncey/go/src/github.com/deis/workflow-cli/parser/keys.go:87 +0x1d7
github.com/deis/workflow-cli/parser.Keys(0xc820088010, 0x1, 0x1, 0x0, 0x0)
	/Users/jonathanchauncey/go/src/github.com/deis/workflow-cli/parser/keys.go:24 +0xbc
main.Command(0xc820088010, 0x1, 0x1, 0x18e0a5)
	/Users/jonathanchauncey/go/src/github.com/deis/workflow-cli/deis.go:105 +0x108a
main.main()
	/Users/jonathanchauncey/go/src/github.com/deis/workflow-cli/deis.go:17 +0x60

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1
```

### PR DESCRIPTION
**Pull request by Joshua-Anderson**
Fixes #91

### CODE REVIEW

**Comment by bacongobbler** [comment]
woudl you mind throwing in some unit tests for this?

**Comment by arschles** [comment]
besides @bacongobbler's above comment, code LGTM. marking as such, assuming a test is added

**Comment by Joshua-Anderson** [comment]
I added some tests, but it required redesigning the function so I could inject an `io.Reader` to mock the standard input.

File: cmd/keys_test.go [hunk 1]
```diff
@@ -121,3 +123,37 @@ func TestListKeys(t *testing.T) {
...
+		chooseKeyCases{
+			Reader:   strings.NewReader("a"),
+			Expected: "a is not a valid integer",
+		},
+	}
+
+	for _, check := range checks {
+		_, err := chooseKey(testKeys, check.Reader)
+
+		if err.Error() != check.Expected {
```
Comment (arschles) [inline cmd/keys_test.go]: tiny nit: is it possible to check on the actual error type or value, instead of the error string?

**Comment by arschles** [comment]
@Joshua-Anderson looks good - thanks for adding those tests!

**Pull request merged**